### ISSUE TITLE
Support CORS

### ISSUE THREAD

**Issue by tarrencev** [eqlabs/pathfinder#784]
It would be useful to support configuring CORS headers for the pathfinder http server. Other nodes like geth provide this. Currently we have a reverse proxy adding the headers which adds a bit of complexity.

It looks like it should be pretty straight forward with jsonrpsee: https://github.com/paritytech/jsonrpsee/blob/3a159638ed6c8ad4c80ad80bea74ef1d13f2f503/examples/examples/cors_server.rs

**Comment by CHr15F0x** [comment]
I think the latest jsonrpsee supports CORS by default. We've got a branch where I actually did update to the latest version. I think your request is a good argument for "unparking" that work. This will require changing the way we handle paths internally for different api versions (abandoning the fork and using tower middleware instead) but that part is already partially ready so I'd just need to polish it.

I'm fresh after the Christmas / NY break so lemme get back to you with a coarse ETA once I discuss priorities with the rest of the team :+1: .
